


<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An internal resource explaining and documenting lab pipelines">
      
      
        <link rel="canonical" href="https://zamanianlab.org/ZamanianLabDocs/pipelines_server/">
      
      
        <meta name="author" content="Zamanian Lab">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-5.5.12">
    
    
      
        <title>Server Pipelines - Zamanian Lab Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.4dd2dd8d.min.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/palette.6a5ad368.min.css">
      
      
        
        
        <meta name="theme-color" content="#4051b5">
      
    
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
    
    
  </head>
  
  
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="indigo" data-md-color-accent="teal">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#server-pipelines" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="https://zamanianlab.org/ZamanianLabDocs" title="Zamanian Lab Documentation" class="md-header-nav__button md-logo" aria-label="Zamanian Lab Documentation">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            Zamanian Lab Documentation
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Server Pipelines
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/zamanianlab/ZamanianLabDocs/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    zamanianlab/ZamanianLabDocs
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
        
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://zamanianlab.org/ZamanianLabDocs" title="Zamanian Lab Documentation" class="md-nav__button md-logo" aria-label="Zamanian Lab Documentation">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Zamanian Lab Documentation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/zamanianlab/ZamanianLabDocs/" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    zamanianlab/ZamanianLabDocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../labmanual/" title="Lab Manual" class="md-nav__link">
      Lab Manual
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../protocols/protocols/" title="Lab Protocols" class="md-nav__link">
      Lab Protocols
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../labsheets/" title="Lab Sheets" class="md-nav__link">
      Lab Sheets
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Computational Guide
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Computational Guide" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        <span class="md-nav__icon md-icon"></span>
        Computational Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_overview/" title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_local/" title="Local Environment" class="md-nav__link">
      Local Environment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_github/" title="GitHub" class="md-nav__link">
      GitHub
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_conda/" title="Conda" class="md-nav__link">
      Conda
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_homebrew/" title="Homebrew" class="md-nav__link">
      Homebrew
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" checked>
    
    <label class="md-nav__link" for="nav-6">
      Pipelines
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Pipelines" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        <span class="md-nav__icon md-icon"></span>
        Pipelines
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../pipelines_overview/" title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pipelines_local/" title="Local Pipelines" class="md-nav__link">
      Local Pipelines
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Server Pipelines
        <span class="md-nav__icon md-icon"></span>
      </label>
    
    <a href="./" title="Server Pipelines" class="md-nav__link md-nav__link--active">
      Server Pipelines
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#a-lab-storage" class="md-nav__link">
    A. Lab Storage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#b-center-for-high-throughput-computing-chtc" class="md-nav__link">
    B. Center for High-throughput Computing (CHTC)
  </a>
  
    <nav class="md-nav" aria-label="B. Center for High-throughput Computing (CHTC)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-htc-system" class="md-nav__link">
    The HTC System
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deploying-pipelines" class="md-nav__link">
    Deploying Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#c-docker" class="md-nav__link">
    C. Docker
  </a>
  
    <nav class="md-nav" aria-label="C. Docker">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#building-docker-images" class="md-nav__link">
    Building Docker Images
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#testing-docker-pipelines" class="md-nav__link">
    Testing Docker Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../editing/" title="Editing" class="md-nav__link">
      Editing
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#a-lab-storage" class="md-nav__link">
    A. Lab Storage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#b-center-for-high-throughput-computing-chtc" class="md-nav__link">
    B. Center for High-throughput Computing (CHTC)
  </a>
  
    <nav class="md-nav" aria-label="B. Center for High-throughput Computing (CHTC)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-htc-system" class="md-nav__link">
    The HTC System
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deploying-pipelines" class="md-nav__link">
    Deploying Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#c-docker" class="md-nav__link">
    C. Docker
  </a>
  
    <nav class="md-nav" aria-label="C. Docker">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#building-docker-images" class="md-nav__link">
    Building Docker Images
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#testing-docker-pipelines" class="md-nav__link">
    Testing Docker Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/zamanianlab/ZamanianLabDocs/edit/master/docs/pipelines_server.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  
                
                
                <h1 id="server-pipelines">Server Pipelines</h1>
<p>Our lab has access to powerful computing resources and support through the <a href="http://chtc.cs.wisc.edu/">Center for High-Throughput Computing (CHTC)</a>. We also have access to virtual servers through the UW <a href="https://www.biotech.wisc.edu/services/brc">Bioinformatics Research Center (BRC)</a>, attached to sequencing data generated by the <a href="https://www.biotech.wisc.edu/">UW Biotech Center</a>. Our core bioinformatics and image processing pipelines will be deployed through CHTC servers. All pipelines will be maintained on <a href="http://www.github.com">GitHub</a> and associated with <a href="http://www.docker.com">Docker</a> environments to ensure reproducibility. Many of our pipelines will use <a href="https://www.nextflow.io/">Nextflow</a>.</p>
<p>In general, pipelines will be run in three steps:</p>
<ul>
<li><strong>Staging:</strong> input files will be transferred to the CHTC server from lab storage</li>
<li><strong>Pipeline:</strong> files will be processed using established pipelines</li>
<li><strong>Output:</strong> desired outputs will be transferred from the CHTC server to lab storage</li>
</ul>
<h2 id="a-lab-storage">A. Lab Storage</h2>
<p>Large data sets (raw and processed) will be stored in locations with reliable backup.</p>
<ul>
<li>
<p>BRC (UW Biotech Center)</p>
<p>Sequencing data generated by the UW Biotechnology Center is delivered to their BRC servers. These data can be accessed in <a href="https://www.biotech.wisc.edu/services/brc/data-access">numerous ways</a>.</p>
</li>
<li>
<p>UW ResearchDrive</p>
<p><a href="https://it.wisc.edu/services/researchdrive">ResearchDrive</a> provides 5 TB (expandable) of secure backed-up storage. Lab members will have access to the lab ResearchDrive using their UW net-id and password. Our sequencing data from the BRC is automatically moved into ResearchDrive. We will also use ResearchDrive to store imaging and all other large data types. General instructions on how to connect to and transfer data in and out of ResearchDrive are provided <a href="https://kb.wisc.edu/researchdata/internal/page.php?id=93998">here</a>. Data will be transferred from ResearchDrive to the CHTC for pipeline-based processing. Outputs will be transferred from the CHTC back to ResearchDrive for long-term storage. Subsets of outputs will be transferred into our lab <a href="http://www.box.com">Box</a> account for post-processing, analysis and plotting. Curent ResearchDrive directory structure is shown below:</p>
<div class="highlight"><pre><span></span><code>/
├── ImageXpress/                          [IX storage]
|   └── raw/                              [Raw exported data]
|   └── metadata/                         [Experiment metadata]    
|   └── proc/                             [CHTC-processed data]
|  
└── UWBC-Dropbox/                         [Auto-deposited data from the UWBC]
|   └── Bioinformatics Resource Center/   [Sequencing Data]
|   └── DNA Sequencing Sanger/            [Sanger Data]
|
└── Box/                                  [Box backup]
|
└── External/                             [External data]
</code></pre></div>

<p>Instructions can be found for <a href="https://kb.wisc.edu/researchdata/internal/page.php?id=93998#connect">connecting</a> to and <a href="https://kb.wisc.edu/researchdata/internal/page.php?id=93998#transferdata">transferring</a> files in and out of ResearchDrive. Files can be transferred into the mounted ResearchDrive using a number of <a href="https://kb.wisc.edu/researchdata/96641">approaches</a>, including simple drag-and-drop or command-line <code>rsync</code> or <code>cp</code>.</p>
<p>Example on a Mac OS X system:</p>
<div class="highlight"><pre><span></span><code>Finder &gt; Go &gt; &quot;Connect to server... &gt;
smb://research.drive.wisc.edu/mzamanian

rsync -rltv ~/Desktop/Data/[dir] /Volumes/mzamanian/ImageXpress/raw/
</code></pre></div>

</li>
<li>
<p>SVM Research Data Storage</p>
<p>SVM PIs will soon have access to secure and backed-up research data storage through the school, stocked initially with 335 TB of storage capacity. Once this storage solution is in place, we will migrate from UW ResearchDrive to the on-premise storage solution.</p>
</li>
</ul>
<h2 id="b-center-for-high-throughput-computing-chtc">B. Center for High-throughput Computing (CHTC)</h2>
<p>Consult official <a href="http://chtc.cs.wisc.edu/">CHTC</a> and <a href="https://research.cs.wisc.edu/htcondor/">HTCondor</a> documentation before getting started. Register for an account using this <a href="http://chtc.cs.wisc.edu/form.shtml">form</a>.</p>
<h3 id="the-htc-system">The HTC System</h3>
<ol>
<li>
<p>Compute Nodes</p>
<p>The CHTC has an extensive set of execute nodes that can be accessed for free. To establish priority access for certain pipelines, our lab has secured a prioritized node that can be accessed on-demand using a designated flag.</p>
<ul>
<li>Typical nodes: 20 cores, 128 GB RAM</li>
<li><a href="http://chtc.cs.wisc.edu/high-memory-jobs.shtml">High-memory nodes</a>: e.g., 80 cores, 4 TB RAM</li>
<li>Dedicated lab node: 40 cores (80 hyperthreading), 512 GB RAM, 3.8 TB HD<br />
    [ Maximum request on lab node: CPU = 80; Memory = 500GB, Disk =  3500GB ]</li>
</ul>
</li>
<li>
<p>Submit nodes</p>
<p>Jobs on the CHTC are deployed from submit nodes. You can <code>ssh</code> into your assigned submit node to run and monitor jobs using your UW net-id and password.</p>
<p><code>ssh {net-id}@submit3.chtc.wisc.edu</code></p>
</li>
<li>
<p>File system</p>
<p>The CHTC does not use a shared file system, but you can request the storage you need for any given job. Each net-id will be associated with a <code>home</code> folder, where you will manage your job submission scripts. Each net-id will also be associated with a <code>staging</code> folder, for transfer of files in and out of the CHTC system.</p>
<div class="highlight"><pre><span></span><code>/
├── home/{net-id}/              [initial quota: 20 GB, submit script dir]
└── staging/{net-id}/           [initial quota: 200 GB | 1000 files]
    └── input/                  [input dir: unprocessed (raw) data]
    └── output/                 [output dir: processed job outputs]
</code></pre></div>

</li>
</ol>
<h3 id="deploying-pipelines">Deploying Pipelines</h3>
<ol>
<li>
<p>Staging input data for processing</p>
<p>Transfer a single folder containing your job input files to the staging input directory. You can run transfer commands from your computer or from the BRC server (sequencing data).</p>
<p><code>scp [dir] {net-id}@transfer.chtc.wisc.edu:/staging/{net-id}/input/</code></p>
<p>More typically, you will be <a href="http://chtc.cs.wisc.edu/transfer-data-researchdrive.shtml">transferring directly</a> between ResearchDrive and CHTC. To transfer a directory from ResearchDrive to the CHTC staging input folder:</p>
<p><details>
<summary> ResearchDrive -&gt; CHTC transfer (Click to Expand)</summary>
<div class="highlight"><pre><span></span><code># log into CHTC staging server and navigate to input folder
ssh {net-id}@transfer.chtc.wisc.edu
cd /staging/{net-id}/input/

# connect to lab ResearchDrive
smbclient -k //research.drive.wisc.edu/mzamanian

# turn off prompting and turn on recursive
smb: \&gt; prompt
smb: \&gt; recurse

# navigate to ResearchDrive dir with raw data (example)
smb: \&gt; cd /ImageXpress/raw/

# transfer raw data folder (example)
smb: \&gt; mget 20200922-p01-NJW_114
</code></pre></div>
</details></p>
</li>
<li>
<p>Creating job submit scripts</p>
<p>CHTC uses HTCondor for job scheduling. Submission files (.sub) should follow lab conventions and be consistent with the CHTC documentation. An example submit script with annotations is shown below. This submit script (Core_RNAseq-nf.sub) loads a pre-defined <a href="https://hub.docker.com/repository/docker/zamanianlab/chtc-rnaseq">Docker environment</a> and runs a bash executable script (Core_RNAseq-nf.sh) with defined arguments (staged data location).</p>
<p>Other options define standard log files, resource requirements (cpu, memory, and hard disk), and transfer of files in/out of <code>home</code>. Avoid transferring large files in/out of <code>home</code>! We transfer in our large data through <code>/staging/{net-id}/input/</code> and we move job output files to <code>/staging/{net-id}/output/</code> within the job executable script to avoid their transfer to <code>home</code> upon job completion. The only files that should be transferred back to <code>home</code> are small log files.</p>
<p><details>
  <summary>Core_RNAseq-nf.sub (Click to Expand)</summary>
  <div class="highlight"><pre><span></span><code># Core_RNAseq-nf.sub
# Input data in /staging/{net-id}/input/$(dir)
# Run: condor_submit Core_RNAseq-nf.sub dir=191211_AHMMC5DMXX netid=mzamanian script=Core_RNAseq-nf.sh

# request Zamanian Lab server
Accounting_Group = PathobiologicalSciences_Zamanian

# load docker image; request execute server with large data staging
universe = docker
docker_image = zamanianlab/chtc-rnaseq:v1
Requirements = (Target.HasCHTCStaging == true)

# executable (/home/{net-id}/) and arguments
executable = $(script)
arguments = $(dir) $(netid)

# log, error, and output files
log = $(dir)_$(Cluster)_$(Process).log
error = $(dir)_$(Cluster)_$(Process).err
output = $(dir)_$(Cluster)_$(Process).out

# transfer files in-out of /home/{net-id}/
transfer_input_files =
should_transfer_files = YES
when_to_transfer_output = ON_EXIT

# memory, disk and CPU requests
request_cpus = 80
request_memory = 500GB
request_disk = 1500GB

# submit 1 job
queue 1
### END
</code></pre></div>
</details></p>
<p>The submit script runs the annotated bash script below on the execute server. This pipeline creates <code>input</code>, <code>work</code>, and <code>output</code> dirs in the loaded Docker environment. It transfers the input data from <code>staging</code> into <code>input</code>, clones a GitHub repo (Nextflow pipeline), and runs a Nextflow command. Nextflow uses <code>work</code> for intermediary processing and spits out any files we have marked for retention into <code>output</code>, which gets transferred back to <code>staging</code>. <code>input</code> and <code>work</code> are deleted before job completion.</p>
<p><details>
  <summary>Core_RNAseq-nf.sh (Click to Expand)</summary>
  <div class="highlight"><pre><span></span><code>#!/bin/bash

# set home () and mk dirs
export HOME=$PWD
mkdir input work output

# echo core, thread, and memory
echo &quot;CPU threads: $(grep -c processor /proc/cpuinfo)&quot;
grep &#39;cpu cores&#39; /proc/cpuinfo | uniq
echo $(free -g)

# transfer input data from staging ($1 is ${dir} and $2 is ${netid} from args)
cp -r /staging/$2/input/$1 input

# clone nextflow git repo
git clone https://github.com/zamanianlab/Core_RNAseq-nf.git

# run nextflow
export NXF_OPTS=&#39;-Xms1g -Xmx8g&#39;
nextflow run Core_RNAseq-nf/WB-pe.nf -w work -c Core_RNAseq-nf/chtc.config \
  --dir $1 --star --release &quot;WBPS14&quot; --species &quot;brugia_malayi&quot; --prjn &quot;PRJNA10729&quot; --rlen &quot;150&quot;

# rm files you don&#39;t want transferred back to /home/{net-id}
rm -r work
rm -r input

# remove staging output folder if there from previous run
rm -r /staging/$2/output/$1

# mv large output files to staging output folder; avoid their transfer back to /home/{net-id}
mv output/$1/ /staging/$2/output/
</code></pre></div>
</details></p>
</li>
<li>
<p>Submitting and managing jobs</p>
<p>Submit job from submit node using <code>condor_submit</code>,</p>
<p><code>condor_submit Core_RNAseq-nf.sub dir=191211_AHMMC5DMXX script=Core_RNAseq-nf.sh</code></p>
<p><details>
  <summary>Other useful commands for monitoring and managing jobs (Click to Expand)</summary>
    <div class="highlight"><pre><span></span><code># check on job status
  condor_q

# remove a specific job
  condor_rm [job id]

# remove all jobs for user
  condor_rm $USER

# interative shell to running job on remote machine
  condor_ssh_to_job [job id]
  exit
</code></pre></div>
</details></p>
</li>
<li>
<p>Transferring output data</p>
<p>To transfer your job output folder from the CHTC staging output directory to your local computer.</p>
<p><code>scp -r {net-id}@transfer.chtc.wisc.edu:/staging/{net-id}/output/[dir] .</code></p>
<p>To transfer your job output directly from the CHTC staging output directory to ResearchDrive.</p>
<p><details>
<summary> ResearchDrive -&gt; CHTC transfer (Click to Expand)</summary>
<div class="highlight"><pre><span></span><code># log into CHTC staging server and navigate to output folder
ssh {net-id}@transfer.chtc.wisc.edu
cd /staging/{net-id}/output/

# connect to lab ResearchDrive
smbclient -k //research.drive.wisc.edu/mzamanian

# turn off prompting and turn on recursive
smb: \&gt; prompt
smb: \&gt; recurse

# navigate to ResearchDrive dir for processed data (example)
smb: \&gt; cd /ImageXpress/proc/

# transfer output data folder (example)
smb: \&gt; mput 20200922-p01-NJW_114
</code></pre></div>
</details></p>
</li>
</ol>
<h2 id="c-docker">C. Docker</h2>
<p>We will user Docker to establish consistent environments (containers) for our established pipelines. We will maintain Docker images on <a href="https://hub.docker.com/orgs/zamanianlab">Docker Hub</a> under the organization name 'zamanianlab'. These images can be directly loaded from Docker Hub in our CHTC submit scripts. The Dockerfiles used to create these images should be maintained in our <a href="https://github.com/zamanianlab/Docker Install">GitHub Docker Repo</a>. Install <a href="https://docs.docker.com/docker-for-mac/install/">Docker Desktop for Mac</a> and create a Dockerhub account to be associated with our organization Docker Hub (zamanianlab).</p>
<h3 id="building-docker-images">Building Docker Images</h3>
<ol>
<li>
<p>Create a lab Docker Hub repo (e.g., zamanianlab/chtc-rnaseq)</p>
</li>
<li>
<p>Create Dockerfile and auxillary (e.g., yaml) files in a folder with the repo name in the <a href="https://github.com/zamanianlab/Docker Install">Docker GitHub repo</a>.</p>
<p>The Dockerfile provides instructions to build a Docker image. In this case, we are starting with the official miniconda Docker image and then installing necessary conda packages into this image. You can search for existing Docker images on <a href="https://hub.docker.com/orgs/zamanianlab">Docker Hub</a> to build on, instead of starting from scratch.</p>
<p><details>
  <summary>Dockerfile (Click to Expand)</summary>
  <div class="highlight"><pre><span></span><code>FROM continuumio/miniconda3
MAINTAINER mzamanian@wisc.edu

# install (nf tracing)
RUN apt-get update &amp;&amp; apt-get install -y procps

# install conda packages
COPY conda_env.yml .
RUN \
   conda env update -n root -f conda_env.yml \
&amp;&amp; conda clean -a
</code></pre></div>
</details></p>
<p>yml file containing <code>conda</code> packages to be installed. You can search for packages on <a href="https://anaconda.org/">Anaconda cloud</a>.</p>
<p><details>
  <summary>conda_env.yml (Click to Expand)</summary>
  <div class="highlight"><pre><span></span><code>conda_env.yaml
  name: rnaseq-nf

  channels:
    - bioconda
    - conda-forge
    - defaults

  dependencies:
    - python=3.8.5
    - nextflow=20.07.1
    - bwa=0.7.17
    - hisat2=2.2.1
    - stringtie=2.1.2
    - fastqc=0.11.9
    - multiQC=1.9
    - fastp=0.20.1
    - bedtools=2.29.2
    - bedops=2.4.39
    - sambamba=0.7.0
    - samtools=1.9
    - picard=2.20.6
    - bcftools=1.9
    - snpeff=4.3.1t
    - mrbayes=3.2.7
    - trimal=1.4.1
    - mafft=7.471
    - muscle=3.8.1551
    - seqtk=1.3
    - raxml=8.2.12
    - htseq=0.12.4
    - mirdeep2=2.0.1.2
</code></pre></div>
</details></p>
</li>
<li>
<p>Build Docker image</p>
<div class="highlight"><pre><span></span><code>cd [/path/to/Dockerfile]
docker build -t zamanianlab/chtc-rnaseq .
</code></pre></div>

</li>
<li>
<p>Test Docker image interactively</p>
<div class="highlight"><pre><span></span><code>docker run -it --rm=TRUE zamanianlab/chtc-rnaseq /bin/bash
ctrl+D to exit
</code></pre></div>

</li>
<li>
<p>Push Docker image to Docker Hub</p>
<div class="highlight"><pre><span></span><code>docker push zamanianlab/chtc-rnaseq
</code></pre></div>

<p><details>
  <summary>Some useful Docker commands (Click to Expand)</summary>
    <div class="highlight"><pre><span></span><code># list docker images
  docker image ls (= docker images)

# remove images
  docker rmi [image]

## remove all docker containers
# run first because images are attached to containers
  docker rm -f $(docker ps -a -q)
# remove every Docker image
  docker rmi -f $(docker images -q)
</code></pre></div>
</details></p>
</li>
</ol>
<h3 id="testing-docker-pipelines">Testing Docker Pipelines</h3>
<p>Before deploying a new pipeline on large datasets, test the pipeline using subsampled data. You can test locally with subsampled data, on the CHTC server with subsampled data, and finally, run the pipeline on the CHTC server with your full dataset. An example is provided below, using RNAseq data.</p>
<ol>
<li>
<p>First, subsample your data:</p>
<div class="highlight"><pre><span></span><code>...
</code></pre></div>

</li>
<li>
<p>Run Docker container locally</p>
<div class="highlight"><pre><span></span><code>docker run -it --rm=TRUE zamanianlab/chtc-rnaseq /bin/bash
</code></pre></div>

</li>
<li>
<p>Simulate the steps in your submit scripts</p>
<p><details>
<summary>Running commands in local Docker container (Click to Expand)</summary>
<div class="highlight"><pre><span></span><code># set home to working directory
export HOME=$PWD

# make input, work, and output directories for nextflow
mkdir input work outputs

# clone GitHub repo that contains pipeline in development
git clone https://github.com/zamanianlab/Core_RNAseq-nf.git

# transfer sub-sampled files from CHTC staging into your input folder
scp -r mzamanian@transfer.chtc.wisc.edu:/staging/mzamanian/input/191211_AHMMC5DMXX/ input

# run nextflow command using chtc-local.config matched to your hardware specs
nextflow run Core_RNAseq-nf/WB-pe.nf -w work -c Core_RNAseq-nf/chtc-local.config --dir &quot;191211_AHMMC5DMXX&quot; --release &quot;WBPS14&quot; --species &quot;brugia_malayi&quot; --prjn &quot;PRJNA10729&quot; --rlen &quot;150&quot;
</code></pre></div>
</details></p>
</li>
<li>
<p>Make changes to your GitHub pipeline, <code>push</code> those changes to GitHub, <code>pull</code> those changes to your local container, and re-run the Nextflow command until the pipeline is behaving as expected.</p>
</li>
</ol>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../pipelines_local/" title="Local Pipelines" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Local Pipelines
              </div>
            </div>
          </a>
        
        
          <a href="../editing/" title="Editing" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Editing
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
  <div class="md-footer-social">
    
      
      
        
        
      
      <a href="https://github.com/zamanianlab" target="_blank" rel="noopener" title="github.com" class="md-footer-social__link">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 480 512"><path d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"/></svg>
      </a>
    
      
      
        
        
      
      <a href="https://twitter.com/zamanian_" target="_blank" rel="noopener" title="twitter.com" class="md-footer-social__link">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
      </a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/vendor.3636a4ec.min.js"></script>
      <script src="../assets/javascripts/bundle.e9fe3281.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents"}</script>
      
      <script>
        app = initialize({
          base: "..",
          features: ["instant"],
          search: Object.assign({
            worker: "../assets/javascripts/worker/search.5eca75d3.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
        <script src="../javascripts/extra.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      
    
  </body>
</html>