



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="An internal resource explaining and documenting lab pipelines">
      
      
        <link rel="canonical" href="https://zamanianlab.org/ZamanianLabDocs/pipelines_server/">
      
      
        <meta name="author" content="Zamanian Lab">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../images/dna.svg">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-4.6.3">
    
    
      
        <title>Server Pipelines - Zamanian Lab Docs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.adb8469c.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.a8b3c06d.css">
      
      
        
        
        <meta name="theme-color" content="">
      
    
    
      <script src="../assets/javascripts/modernizr.86422ebf.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Lato","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../stylesheets/extra_colors.css">
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="black" data-md-color-accent="red">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#server-pipelines" tabindex="0" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://zamanianlab.org/ZamanianLabDocs" title="Zamanian Lab Docs" aria-label="Zamanian Lab Docs" class="md-header-nav__button md-logo">
          
            <img alt="logo" src="../images/dna.svg" width="24" height="24">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Zamanian Lab Docs
            </span>
            <span class="md-header-nav__topic">
              
                Server Pipelines
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" aria-label="search" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/zamanianlab/ZamanianLabDocs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    zamanianlab/ZamanianLabDocs
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://zamanianlab.org/ZamanianLabDocs" title="Zamanian Lab Docs" class="md-nav__button md-logo">
      
        <img alt="logo" src="../images/dna.svg" width="48" height="48">
      
    </a>
    Zamanian Lab Docs
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/zamanianlab/ZamanianLabDocs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    zamanianlab/ZamanianLabDocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../labmanual/" title="Lab Manual" class="md-nav__link">
      Lab Manual
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../protocols/protocols/" title="Lab Protocols" class="md-nav__link">
      Lab Protocols
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../labsheets/" title="Lab Sheets" class="md-nav__link">
      Lab Sheets
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Computational Guide
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Computational Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_overview/" title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_local/" title="Local Environment" class="md-nav__link">
      Local Environment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_github/" title="GitHub" class="md-nav__link">
      GitHub
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_conda/" title="Conda" class="md-nav__link">
      Conda
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../comp_homebrew/" title="Homebrew" class="md-nav__link">
      Homebrew
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" checked>
    
    <label class="md-nav__link" for="nav-6">
      Pipelines
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Pipelines
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../pipelines_overview/" title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pipelines_storage/" title="Data Storage" class="md-nav__link">
      Data Storage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pipelines_local/" title="Local Pipelines" class="md-nav__link">
      Local Pipelines
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Server Pipelines
      </label>
    
    <a href="./" title="Server Pipelines" class="md-nav__link md-nav__link--active">
      Server Pipelines
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#a-center-for-high-throughput-computing-chtc" class="md-nav__link">
    A. Center for High-throughput Computing (CHTC)
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-htc-system" class="md-nav__link">
    The HTC System
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deploying-pipelines" class="md-nav__link">
    Deploying Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#b-docker" class="md-nav__link">
    B. Docker
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#building-docker-images" class="md-nav__link">
    Building Docker Images
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#testing-docker-pipelines" class="md-nav__link">
    Testing Docker Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../resources/" title="Resources" class="md-nav__link">
      Resources
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#a-center-for-high-throughput-computing-chtc" class="md-nav__link">
    A. Center for High-throughput Computing (CHTC)
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-htc-system" class="md-nav__link">
    The HTC System
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deploying-pipelines" class="md-nav__link">
    Deploying Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#b-docker" class="md-nav__link">
    B. Docker
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#building-docker-images" class="md-nav__link">
    Building Docker Images
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#testing-docker-pipelines" class="md-nav__link">
    Testing Docker Pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/zamanianlab/ZamanianLabDocs/edit/master/docs/pipelines_server.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="server-pipelines">Server Pipelines</h1>
<p>Our lab has access to powerful computing resources and support through the <a href="http://chtc.cs.wisc.edu/">Center for High-Throughput Computing (CHTC)</a>. Our core bioinformatics and image processing pipelines will be deployed through CHTC servers. All pipelines will be maintained on <a href="http://www.github.com">GitHub</a> and associated with <a href="http://www.docker.com">Docker</a> environments to ensure reproducibility. Many of our pipelines will use <a href="https://www.nextflow.io/">Nextflow</a>.</p>
<p>In general, pipelines will be run in three steps:</p>
<ul>
<li><strong>Staging:</strong> input files will be transferred to the CHTC server from lab storage</li>
<li><strong>Pipeline:</strong> files will be processed using established pipelines</li>
<li><strong>Output:</strong> desired outputs will be transferred from the CHTC server to lab storage</li>
</ul>
<h2 id="a-center-for-high-throughput-computing-chtc">A. Center for High-throughput Computing (CHTC)</h2>
<p>Consult official <a href="http://chtc.cs.wisc.edu/">CHTC</a> and <a href="https://research.cs.wisc.edu/htcondor/">HTCondor</a> documentation before getting started. Register for an account using this <a href="http://chtc.cs.wisc.edu/form.shtml">form</a>.</p>
<p><img src="../images/chtc_flowchart.png" width="100%"></p>
<h3 id="the-htc-system">The HTC System</h3>
<ol>
<li>
<p>Execute (Compute) nodes</p>
<p>The CHTC has an extensive set of execute nodes. To establish priority access for certain pipelines, our lab has secured a prioritized node that can be accessed on-demand using a designated flag.</p>
<ul>
<li>Typical nodes: 20 cores, 128 GB RAM</li>
<li><a href="http://chtc.cs.wisc.edu/high-memory-jobs.shtml">High-memory nodes</a>: e.g., 80 cores, 4 TB RAM</li>
<li>Dedicated lab node: 40 cores (80 hyperthreading), 512 GB RAM, 3.8 TB HD  </li>
</ul>
</li>
<li>
<p>Submit nodes</p>
<p>Jobs on the CHTC are deployed from submit nodes. You can <code>ssh</code> into our assigned submit node (submit2) to run and monitor jobs using your UW net-id and password.</p>
<p><code>ssh {net-id}@submit2.chtc.wisc.edu</code></p>
<p><strong>Note:</strong> If you correctly updated your <code>~/.bash_profile</code> by following the <a href="../comp_local/">macOS environment setup instructions</a>, then you can use the simple <code>submit</code> to <code>ssh</code> into the node.</p>
</li>
<li>
<p>File system</p>
<p>Each net-id is associated with a <code>home</code> folder, where we manage job submission scripts. Our lab also has a shared <code>staging</code> folder, for transfer of large files in and out of the CHTC system. The CHTC does not use a shared file system, but you can request the storage you need for any given job.</p>
<div class="highlight"><pre><span></span><code>/
├── home/{net-id}/                    [quota: 20 GB, submit script dir]
└── staging/groups/zamanian_group/    [quota: 1 TB | 100 files]
    └── input/                        [input dir: unprocessed (raw) data]
    └── output/                       [output dir: processed job outputs]
    └── WBP.tar.gz                    [permanent storage of WBP data]
</code></pre></div>
</li>
</ol>
<h3 id="deploying-pipelines">Deploying Pipelines</h3>
<ol>
<li>
<p><strong>Staging -</strong> transfer input data for processing (ResearchDrive -&gt; CHTC)</p>
<p>1A. <strong>Globus transfer</strong></p>
<p>In almost all cases, you will use <a href="https://it.wisc.edu/it-projects/globus-research-data-management-project/">Globus</a> to transfer your input data from ResearchDrive to the CHTC staging input folder. Globus is the fastest and most secure transfer method, and allows for transfer from any file system that has a Globus endpoint installed. Most raw data on ResearchDrive is unarchived and uncompressed. However, our pipelines expect a single archived folder (.tar) as input and will deliver a single archived folder as output. Use the workflow below to transfer an unarchived folder on ResearchDrive to CHTC input and archive it after arrival. See the <a href="https://kb.wisc.edu/researchdata/internal/page.php?id=108855">KB</a> for further instructions and necessary preparations for initiating your first transfer.</p>
<p><details>
<summary> Globus transfer</summary></p>
<ol>
<li>Login to the <a href="https://app.globus.org/">Globus web interface</a> with your NetID</li>
<li>If transferring from a personal computer, install and start <a href="https://www.globus.org/globus-connect-personal">Globus Connect Personal</a>.</li>
<li>If transferring from ResearchDrive, first create a kerberos ticket by running the command <code>ssh [netid]@doit-rci-00025.doit.wisc.edu</code>.</li>
<li>In the web interface, set the view to two panels using the icon on the top right.</li>
<li>On one side of the interface, click Collection and choose the desired endpoint (<code>chtc#staging</code>, <code>wisc-drive</code>, or your personal computer).</li>
<li>Choose the other endpoint for the other side of the interface.</li>
<li>Type <code>/staging/groups/zamanian_group/</code> into the Path box of the <code>chtc#staging</code> collection and press Enter (you may be required to login with your NetID again). Use <code>/mnt/researchdrive/mzamanian/</code> for <code>wisc-drive</code>.</li>
<li>Navigate to the desired directories.</li>
<li>Drag and drop files to transfer them; you will receive an email upon transfer completion.</li>
<li>Exit the SSH once finished transferring to/from ResearchDrive.</li>
<li>Login to the transfer server and archive the directories in <code>input/</code> and <code>metadata/</code> with the command <code>tar -c {plate}.tar {plate}</code>.</li>
<li>Delete the original, unarchived directories.</li>
</ol>
</details>
<p>1B. <strong>Command line transfer with smbclient</strong></p>
<p>It is also possible to transfer via smbclient using the terminal. The following code will also archive the data upon arrival.</p>
<p><details>
<summary> ResearchDrive -&gt; CHTC transfer of unarchived raw data folder (archived on arrival)</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into transfer server and navigate to staging input dir</span>
ssh <span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span> /staging/groups/zamanian_group/input/

<span class="c1"># Example of transferring sequencing data</span>
smbclient -k //research.drive.wisc.edu/mzamanian/ -D <span class="s2">&quot;UWBC-Dropbox/Bioinformatics Resource Center&quot;</span> -Tc 201105_AHLVWJDSXY.tar <span class="s2">&quot;201105_AHLVWJDSXY&quot;</span>

<span class="c1"># Example of transferring ImageXpress data</span>
smbclient -k //research.drive.wisc.edu/mzamanian/ -D <span class="s2">&quot;ImageXpress/raw&quot;</span> -Tc <span class="m">20201118</span>-p01-MZ_172.tar <span class="s2">&quot;20201118-p01-MZ_172.tar&quot;</span>
</code></pre></div>
</details></p>
<p><details>
<summary> ResearchDrive -&gt; CHTC transfer of unarchived metadata folder (archived on arrival)</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into transfer server and navigate to staging metadata dir</span>
ssh <span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span> /staging/groups/zamanian_group/metadata/

<span class="c1"># Example of transferring ImageXpress metadata</span>
smbclient -k //research.drive.wisc.edu/mzamanian/ -D <span class="s2">&quot;ImageXpress/metadata&quot;</span> -Tc <span class="m">20201118</span>-p01-MZ_172.tar <span class="s2">&quot;20201118-p01-MZ_172&quot;</span>
</code></pre></div>
</details></p>
<p><strong>Note:</strong> If you correctly updated your <code>~/.bash_profile</code> by following the <a href="../comp_local/">macOS environment setup instructions</a>, then you can use the simple <code>transfer</code> to <code>ssh</code> into the node.</p>
<p>For ImageXpress data, an entire experiment may include &gt;10 plates that will take hours to days to transfer. To facilitate batch transfers, we have included two scripts in the <code>input/</code> and <code>metadata/</code> directories of <code>/staging/groups/zamanian_group/</code> called <code>transfer_images.sh</code> and <code>transfer_metadata.sh</code>. These scripts reference the text file <code>/staging/groups/zamanian_group/plates.txt</code> and will loop through the plate names, sequentially transferring them from ResearchDrive and archiving them upon arrival. Edit <code>plates.txt</code> in the terminal using <code>vi</code> or <code>nano</code>, or edit the file locally and then upload it to <code>/staging/groups/zamanian_group/input/</code> (be sure to include a single plate name on each line with a blank line at the end of the file.) Run <code>sh transfer_metadata.sh</code> and <code>sh transfer_images.sh</code> to initate the transfers.</p>
<p>Both of these scripts will require a continuous ssh connection while they transfer. Transferring metadata will only take a few minutes, but transferring multiple plates of images will take several hours. The transfer process can be sent to the background (allowing the user to close the ssh connection) by using the <code>screen</code> tool. There are a number of helpful tutorials online, but a few sample commands are shown below.</p>
<p><details>
<summary> Background transfer using <code>screen</code></summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into transfer server and navigate to staging input dir</span>
ssh <span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span> /staging/groups/zamanian_group/input/

<span class="c1"># Start a screen named &#39;transfer&#39;</span>
screen -S transfer

<span class="c1"># Initiate the transfer</span>
sh transfer_images.sh

<span class="c1"># Detach from the screen by pressing Ctrl+a and then d</span>
<span class="c1"># Reattach to the screen</span>
screen -r transfer

<span class="c1"># Close the screen</span>
<span class="nb">exit</span>
</code></pre></div>
</details></p>
<p>Rarely, you may have to transfer data from other sources to CHTC staging input. You can run simple transfer commands from your computer:</p>
<p><code>scp [dir] {net-id}@transfer.chtc.wisc.edu:/staging/groups/zamanian_group/input/</code></p>
</li>
<li>
<p><strong>Pipeline -</strong> Submit and manage CHTC jobs</p>
<p>CHTC uses HTCondor for job scheduling. Submission files should follow lab conventions and be consistent with the CHTC documentation. An example submit script with annotations is shown below. This submit script (Core_RNAseq-nf.sub) loads a pre-defined <a href="https://hub.docker.com/repository/docker/zamanianlab/chtc-rnaseq">Docker environment</a> and runs a bash executable script (Core_RNAseq-nf.sh) with defined arguments on the execute node. Other options define log files, resource requirements , and transfer of files in/out of <code>home</code>. Avoid transferring large files in/out of <code>home</code>! We transfer in our large data through <code>/staging/groups/zamanian_group/input/</code> and we move job output files to <code>/staging/groups/zamanian_group/output/</code> within the job executable script to avoid their transfer to <code>home</code> upon job completion. The only files that should be transferred back to <code>home</code> are small log files.</p>
<p><details>
<summary> Example CHTC job submission scripts (.sub / .sh)</summary></p>
<div class="tabbed-set" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><label for="__tabbed_1_1">Core_RNAseq-nf.sub</label><div class="tabbed-content">
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</pre></div></td><td class="code"><div class="highlight"><pre><span></span><code># Core_RNAseq-nf.sub
# Input data in /staging/{net-id}/input/$(dir)
# Run: condor_submit Core_RNAseq-nf.sub dir=191211_AHMMC5DMXX script=Core_RNAseq-nf.sh

# request Zamanian Lab server
Accounting_Group = PathobiologicalSciences_Zamanian

# load docker image; request execute server with staging
universe = docker
docker_image = zamanianlab/chtc-rnaseq:v1
Requirements = (Target.HasCHTCStaging == true)

# executable (/home/{net-id}/) and arguments
executable = $(script)
arguments = $(dir)

# log, error, and output files
log = $(dir)_$(Cluster)_$(Process).log
error = $(dir)_$(Cluster)_$(Process).err
output = $(dir)_$(Cluster)_$(Process).out

# transfer files in-out of /home/{net-id}/
transfer_input_files =
should_transfer_files = YES
when_to_transfer_output = ON_EXIT

# memory, disk and CPU requests
request_cpus = 80
request_memory = 500GB
request_disk = 1500GB

# submit 1 job
queue 1
### END
</code></pre></div>
</td></tr></table>
</div>
<input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><label for="__tabbed_1_2">Core_RNAseq-nf.sh</label><div class="tabbed-content">
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</pre></div></td><td class="code"><div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>

<span class="c1"># set home () and mk dirs</span>
<span class="nb">export</span> <span class="nv">HOME</span><span class="o">=</span><span class="nv">$PWD</span>
mkdir input work output

<span class="c1"># echo core, thread, and memory</span>
<span class="nb">echo</span> <span class="s2">&quot;CPU threads: </span><span class="k">$(</span>grep -c processor /proc/cpuinfo<span class="k">)</span><span class="s2">&quot;</span>
grep <span class="s1">&#39;cpu cores&#39;</span> /proc/cpuinfo <span class="p">|</span> uniq
<span class="nb">echo</span> <span class="k">$(</span>free -g<span class="k">)</span>

<span class="c1"># transfer input data from staging ($1 is ${dir} from args)</span>
cp -r /staging/groups/zamanian_group/input/<span class="nv">$1</span>.tar input
<span class="nb">cd</span> input <span class="o">&amp;&amp;</span> tar -xvf <span class="nv">$1</span>.tar <span class="o">&amp;&amp;</span> rm <span class="nv">$1</span>.tar <span class="o">&amp;&amp;</span> mv */*/* <span class="nv">$1</span> <span class="o">&amp;&amp;</span> <span class="nb">cd</span> ..

<span class="c1"># clone nextflow git repo</span>
git clone https://github.com/zamanianlab/Core_RNAseq-nf.git

<span class="c1"># run nextflow command</span>
<span class="nb">export</span> <span class="nv">NXF_OPTS</span><span class="o">=</span><span class="s1">&#39;-Xms1g -Xmx8g&#39;</span>
nextflow run Core_RNAseq-nf/WB-pe.nf -w work -c Core_RNAseq-nf/chtc.config --dir <span class="nv">$1</span><span class="se">\</span>
   --star --qc --release <span class="s2">&quot;WBPS15&quot;</span> --species <span class="s2">&quot;brugia_malayi&quot;</span> --prjn <span class="s2">&quot;PRJNA10729&quot;</span> --rlen <span class="s2">&quot;150&quot;</span>

<span class="c1"># rm files you don&#39;t want transferred back to /home/{net-id}</span>
rm -r work input

<span class="c1"># tar output folder and delete it</span>
<span class="nb">cd</span> output <span class="o">&amp;&amp;</span> tar -cvf <span class="nv">$1</span>.tar <span class="nv">$1</span> <span class="o">&amp;&amp;</span> rm -r <span class="nv">$1</span> <span class="o">&amp;&amp;</span> <span class="nb">cd</span> ..

<span class="c1"># remove staging output tar if there from previous run</span>
rm -f /staging/groups/zamanian_group/output/<span class="nv">$1</span>.tar

<span class="c1"># mv large output files to staging output folder; avoid their transfer back to /home/{net-id}</span>
mv output/<span class="nv">$1</span>.tar /staging/groups/zamanian_group/output/
</code></pre></div>
</td></tr></table>
</div>
</div>
</details>
<p>Log into submit node to submit a job,</p>
<div class="highlight"><pre><span></span><code>ssh <span class="o">{</span>net-id<span class="o">}</span>@submit2.chtc.wisc.edu

condor_submit Core_RNAseq-nf.sub <span class="nv">dir</span><span class="o">=</span>191211_AHMMC5DMXX <span class="nv">script</span><span class="o">=</span>Core_RNAseq-nf.sh
</code></pre></div>
<p><details>
  <summary>Other useful commands for monitoring and managing jobs (Click to Expand)</summary>
    <div class="highlight"><pre><span></span><code><span class="c1"># check on job status</span>
  condor_q

<span class="c1"># remove a specific job</span>
  condor_rm <span class="o">[</span>job id<span class="o">]</span>

<span class="c1"># remove all jobs for user</span>
  condor_rm <span class="nv">$USER</span>

<span class="c1"># interative shell to running job on remote machine</span>
  condor_ssh_to_job <span class="o">[</span>job id<span class="o">]</span>
  <span class="nb">exit</span>
</code></pre></div>
</details></p>
</li>
<li>
<p><strong>Output -</strong> transfer output data (CHTC -&gt; ResearchDrive)</p>
<p>To transfer your job output folder from the CHTC staging output directory to Research Drive:</p>
<p><details>
<summary> CHTC -&gt; ResearchDrive transfer (Click to Expand)</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># log into CHTC staging server and navigate to output folder</span>
ssh <span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span> /staging/groups/zamanian_group/output/

<span class="c1"># connect to lab ResearchDrive</span>
smbclient -k //research.drive.wisc.edu/mzamanian

<span class="c1"># turn off prompting and turn on recursive</span>
smb: <span class="se">\&gt;</span> prompt
smb: <span class="se">\&gt;</span> recurse

<span class="c1"># navigate to ResearchDrive dir for processed data (example)</span>
smb: <span class="se">\&gt;</span> <span class="nb">cd</span> /ImageXpress/proc/

<span class="c1"># transfer output data folder (example)</span>
smb: <span class="se">\&gt;</span> mput <span class="m">20201119</span>-p01-MZ_200.tar
</code></pre></div>
</details></p>
<p>Output data can also be transferred to your computer directly from the CHTC (as shown in the command below), or from the mounted ResearchDrive if the data have already been moved to ResearchDrive.</p>
<p><code>scp -r {net-id}@transfer.chtc.wisc.edu:/staging/groups/zamanian_group/output/[dir] .</code></p>
</li>
</ol>
<h2 id="b-docker">B. Docker</h2>
<p>We will user Docker to establish consistent environments (containers) for our established pipelines. We will maintain Docker images on <a href="https://hub.docker.com/orgs/zamanianlab">Docker Hub</a> under the organization name 'zamanianlab'. These images can be directly loaded from Docker Hub in our CHTC submit scripts. The Dockerfiles used to create these images should be maintained in our <a href="https://github.com/zamanianlab/Docker Install">GitHub Docker Repo</a>. Install <a href="https://docs.docker.com/docker-for-mac/install/">Docker Desktop for Mac</a> and create a Dockerhub account to be associated with our organization Docker Hub (zamanianlab).</p>
<h3 id="building-docker-images">Building Docker Images</h3>
<ol>
<li>
<p>Create a lab Docker Hub repo (e.g., zamanianlab/chtc-rnaseq)</p>
</li>
<li>
<p>Create Dockerfile and auxillary (e.g., yaml) files in a folder with the repo name in the <a href="https://github.com/zamanianlab/Docker Install">Docker GitHub repo</a>.</p>
<p>The Dockerfile provides instructions to build a Docker image. In this case, we are starting with the official miniconda Docker image and then installing necessary conda packages into this image. You can search for existing Docker images on <a href="https://hub.docker.com/orgs/zamanianlab">Docker Hub</a> to build on, instead of starting from scratch.</p>
<p><details>
  <summary>Dockerfile (Click to Expand)</summary>
  <div class="highlight"><pre><span></span><code>FROM continuumio/miniconda3
MAINTAINER mzamanian@wisc.edu

# install (nf tracing)
RUN apt-get update &amp;&amp; apt-get install -y procps

# install conda packages
COPY conda_env.yml .
RUN \
   conda env update -n root -f conda_env.yml \
&amp;&amp; conda clean -a
</code></pre></div>
</details></p>
<p>The following yml file lists <code>conda</code> packages to be installed. You can search for packages on <a href="https://anaconda.org/">Anaconda cloud</a>.</p>
<p><details>
  <summary>conda_env.yml (Click to Expand)</summary>
  <div class="highlight"><pre><span></span><code>conda_env.yaml
  name: rnaseq-nf

  channels:
    - bioconda
    - conda-forge
    - defaults

  dependencies:
    - python=3.8.5
    - nextflow=20.07.1
    - bwa=0.7.17
    - hisat2=2.2.1
    - stringtie=2.1.2
    - fastqc=0.11.9
    - multiQC=1.9
    - fastp=0.20.1
    - bedtools=2.29.2
    - bedops=2.4.39
    - sambamba=0.7.0
    - samtools=1.9
    - picard=2.20.6
    - bcftools=1.9
    - snpeff=4.3.1t
    - mrbayes=3.2.7
    - trimal=1.4.1
    - mafft=7.471
    - muscle=3.8.1551
    - seqtk=1.3
    - raxml=8.2.12
    - htseq=0.12.4
    - mirdeep2=2.0.1.2
</code></pre></div>
</details></p>
</li>
<li>
<p>Build Docker image</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> <span class="o">[</span>/path/to/Dockerfile<span class="o">]</span>
docker build -t zamanianlab/chtc-rnaseq .
</code></pre></div>
</li>
<li>
<p>Test Docker image interactively</p>
<div class="highlight"><pre><span></span><code>docker run -it --rm<span class="o">=</span>TRUE zamanianlab/chtc-rnaseq /bin/bash
ctrl+D to <span class="nb">exit</span>
</code></pre></div>
</li>
<li>
<p>Push Docker image to Docker Hub</p>
<div class="highlight"><pre><span></span><code>docker push zamanianlab/chtc-rnaseq
</code></pre></div>
<p><details>
  <summary>Some useful Docker commands (Click to Expand)</summary>
    <div class="highlight"><pre><span></span><code><span class="c1"># list docker images</span>
  docker image ls <span class="o">(=</span> docker images<span class="o">)</span>

<span class="c1"># remove images</span>
  docker rmi <span class="o">[</span>image<span class="o">]</span>

<span class="c1">## remove all docker containers</span>
<span class="c1"># run first because images are attached to containers</span>
  docker rm -f <span class="k">$(</span>docker ps -a -q<span class="k">)</span>
<span class="c1"># remove every Docker image</span>
  docker rmi -f <span class="k">$(</span>docker images -q<span class="k">)</span>
</code></pre></div>
</details></p>
</li>
</ol>
<h3 id="testing-docker-pipelines">Testing Docker Pipelines</h3>
<p>Before deploying a new pipeline on large datasets, test the pipeline using subsampled data. You can test locally with subsampled data, on the CHTC server with subsampled data, and finally, run the pipeline on the CHTC server with your full dataset. An example is provided below, using RNAseq data.</p>
<ol>
<li>
<p>First, subsample your data into a more manageable size and store it in the staging <code>subsampled</code> folder.</p>
</li>
<li>
<p>Run Docker container locally</p>
<div class="highlight"><pre><span></span><code>docker run -it --rm<span class="o">=</span>TRUE zamanianlab/chtc-rnaseq:v2 /bin/bash
</code></pre></div>
</li>
<li>
<p>Simulate the steps in your submit scripts</p>
<p><details>
<summary>Running commands in local Docker container (Click to Expand)</summary>
<div class="highlight"><pre><span></span><code># set home to working directory
export HOME=$PWD

# make input, work, and output directories for nextflow
mkdir input work outputs

# clone GitHub repo that contains pipeline in development
git clone https://github.com/zamanianlab/Core_RNAseq-nf.git

# transfer sub-sampled files from CHTC staging into your input folder
scp -r {net-id}@transfer.chtc.wisc.edu:/staging/groups/zamanian_group/subsampled/191211_AHMMC5DMXX.tar input

# run your pipeline commands

# example of a nextflow command using chtc-local.config matched to your hardware specs
nextflow run Core_RNAseq-nf/WB-pe.nf -w work -c Core_RNAseq-nf/chtc-local.config --dir &quot;191211_AHMMC5DMXX&quot; --release &quot;WBPS14&quot; --species &quot;brugia_malayi&quot; --prjn &quot;PRJNA10729&quot; --rlen &quot;150&quot;
</code></pre></div>
</details></p>
</li>
<li>
<p>Make changes to your GitHub pipeline, <code>push</code> those changes to GitHub, <code>pull</code> those changes to your local container, and re-run the Nextflow command until the pipeline is behaving as expected.</p>
</li>
</ol>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../pipelines_local/" title="Local Pipelines" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Local Pipelines
              </span>
            </div>
          </a>
        
        
          <a href="../resources/" title="Resources" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Resources
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org" target="_blank" rel="noopener">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.c33a9706.js"></script>
      
      <script>app.initialize({version:"1.1",url:{base:".."}})</script>
      
        <script src="../javascripts/extra.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      
    
  </body>
</html>