
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An internal resource explaining and documenting lab pipelines">
      
      
        <meta name="author" content="Zamanian Lab">
      
      
        <link rel="canonical" href="https://zamanianlab.org/ZamanianLabDocs/pipelines_server/">
      
      
        <link rel="prev" href="../pipelines_local/">
      
      
        <link rel="next" href="../pipelines_docker/">
      
      
      <link rel="icon" href="../images/dna.svg">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.30">
    
    
      
        <title>Server Pipelines - Zamanian Lab Docs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.3cba04c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Lato";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra_colors.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="red">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#chtc-overview" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Zamanian Lab Docs" class="md-header__button md-logo" aria-label="Zamanian Lab Docs" data-md-component="logo">
      
  <img src="../images/dna.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Zamanian Lab Docs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Server Pipelines
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/zamanianlab/ZamanianLabDocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    zamanianlab/ZamanianLabDocs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Zamanian Lab Docs" class="md-nav__button md-logo" aria-label="Zamanian Lab Docs" data-md-component="logo">
      
  <img src="../images/dna.svg" alt="logo">

    </a>
    Zamanian Lab Docs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/zamanianlab/ZamanianLabDocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    zamanianlab/ZamanianLabDocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../labmanual/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Manual
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Protocols
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../labgdrive/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GDrive
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computational Guide
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Computational Guide
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp_overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp_local/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Local Environment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp_storage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Storage
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp_github/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GitHub
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp_conda/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Conda
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp_homebrew/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Homebrew
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Pipelines
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Pipelines
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pipelines_overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pipelines_local/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Local Pipelines
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Server Pipelines
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Server Pipelines
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#chtc-overview" class="md-nav__link">
    <span class="md-ellipsis">
      CHTC Overview
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pipelines" class="md-nav__link">
    <span class="md-ellipsis">
      Pipelines
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Pipelines">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-staging-researchdrive-chtc" class="md-nav__link">
    <span class="md-ellipsis">
      1. Staging (ResearchDrive &rarr; CHTC)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-submit-manage-chtc-jobs" class="md-nav__link">
    <span class="md-ellipsis">
      2. Submit (manage CHTC jobs)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-output-chtc-researchdrive" class="md-nav__link">
    <span class="md-ellipsis">
      3. Output (CHTC &rarr; ResearchDrive)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pipelines_docker/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Docker
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pipelines_wrmxpress/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    wrmXpress Pipelines
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#chtc-overview" class="md-nav__link">
    <span class="md-ellipsis">
      CHTC Overview
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pipelines" class="md-nav__link">
    <span class="md-ellipsis">
      Pipelines
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Pipelines">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-staging-researchdrive-chtc" class="md-nav__link">
    <span class="md-ellipsis">
      1. Staging (ResearchDrive &rarr; CHTC)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-submit-manage-chtc-jobs" class="md-nav__link">
    <span class="md-ellipsis">
      2. Submit (manage CHTC jobs)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-output-chtc-researchdrive" class="md-nav__link">
    <span class="md-ellipsis">
      3. Output (CHTC &rarr; ResearchDrive)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


  <h1>Server Pipelines</h1>

<h2 id="chtc-overview">CHTC Overview</h2>
<p>Our lab has access to powerful computing resources and support through the <a href="https://chtc.cs.wisc.edu/">Center for High-Throughput Computing (CHTC)</a>. Our core bioinformatics and image processing pipelines will be deployed through CHTC servers. All pipelines will be maintained on <a href="https://www.github.com">GitHub</a> and associated with <a href="https://www.docker.com">Docker</a> environments to ensure reproducibility. Many of our pipelines will use <a href="https://www.nextflow.io/">Nextflow</a>. Consult official <a href="https://chtc.cs.wisc.edu/">CHTC</a> documentation before getting started. </p>
<p>Register for an account using this <a href="https://chtc.cs.wisc.edu/uw-research-computing/form">form</a> to access CHTC and our prioritized lab server node.</p>
<ul>
<li>
<p><strong>Execute (Compute) nodes:</strong> The CHTC has an extensive set of execute nodes. Our lab has a prioritized node that can be accessed on-demand using a designated flag.</p>
<details class="example">
<summary>CHTC Compute Nodes</summary>
<ul>
<li>Dedicated lab node: 40 cores (80 hyperthreading), 512 GB RAM, 3.8 TB HD  </li>
<li>Typical nodes: 20 cores, 128 GB RAM</li>
<li><a href="https://chtc.cs.wisc.edu/uw-research-computing/high-memory-jobs">High-memory nodes</a>: e.g., 80 cores, 4 TB RAM</li>
<li><a href="https://chtc.cs.wisc.edu/uw-research-computing/gpu-lab.html">GPUs</a> for machine learning jobs</li>
</ul>
</details>
</li>
<li>
<p><strong>Submit server:</strong> Jobs on the CHTC are deployed from submit servers. You can <code>ssh</code> into our assigned submit server (ap2002) to run and monitor jobs using your UW net-id and password. If you correctly updated your <code>~/.bash_profile</code> by following the <a href="../comp_local/">environment setup instructions</a>, you can use the simple <code>submit</code> to <code>ssh</code> into the node.</p>
<details>
<summary>Connecting to CHTC submit server</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into submit server</span>
ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@ap2002.chtc.wisc.edu

<span class="c1"># Log into submit server if you have bash profile alias setup</span>
submit
</code></pre></div>
</details>
</li>
<li>
<p><strong>Transfer server:</strong> While movement of smaller files and scripts to CHTC be handled in a number of ways, the transfer of large files to <a href="https://chtc.cs.wisc.edu/uw-research-computing/file-avail-largedata">CHTC Staging</a> should use the dedicated transfer server.</p>
<details>
<summary>Connecting to CHTC transfer server</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into submit server</span>
ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu

<span class="c1"># Log into transfer server if you have bash profile alias setup</span>
transfer
</code></pre></div>
</details>
</li>
<li>
<p><strong>File system:</strong> Each net-id is associated with a <code>home</code> folder, where we manage job submission scripts. We have a shared <code>staging</code> folder for transfer of large files in/out of the CHTC system. The directory structure of the CHTC is shown below.</p>
<div class="highlight"><pre><span></span><code>├── home/{net-id}/                    [quota: 20 GB, submit script dir]
└── staging/groups/zamanian_group/    [quota: 1 TB | 100k files]
    └── input/                        [input dir: unprocessed (raw) data]
    └── metadata/                     [metadata dir: metadata associated with input data]
    └── output/                       [output dir: processed job outputs]
</code></pre></div>
</li>
</ul>
<h2 id="pipelines">Pipelines</h2>
<p>In general, pipelines will be run in three steps:</p>
<ol>
<li><strong>Staging:</strong> large input files will be transferred to CHTC Staging from UW ResearchDrive</li>
<li><strong>Submit:</strong> jobs will be submitted to the CHTC to process files using established pipelines</li>
<li><strong>Output:</strong> desired outputs will be transferred from CHTC to UW ResearchDrive and/or Box</li>
</ol>
<p><img src="../images/chtc_flowchart.png" width="100%"></p>
<h3 id="1-staging-researchdrive-chtc">1. Staging (ResearchDrive &rarr; CHTC)</h3>
<ul>
<li>
<p><strong>Globus transfer</strong>: <a href="https://it.wisc.edu/it-projects/globus-research-data-management-project/">Globus</a> is the easiest and fastest way to transfer large input data from ResearchDrive to the CHTC staging input folder. Globus is also the most secure transfer method, and allows for transfer from any file system that has a Globus endpoint installed. Most raw data on ResearchDrive is unarchived and uncompressed. However, our pipelines expect a single archived folder (.tar) as input and will deliver a single archived folder as output. Use the workflow below to transfer an unarchived folder on ResearchDrive to CHTC input and archive it after arrival. See the <a href="https://kb.wisc.edu/researchdata/internal/page.php?id=108855">KB</a> for further instructions.</p>
<details class="example">
<summary>Globus transfer</summary>
<ol>
<li>Login to the <a href="https://app.globus.org/">Globus web interface</a> with your NetID</li>
<li>If transferring from a personal computer, install and start <a href="https://www.globus.org/globus-connect-personal">Globus Connect Personal</a>.</li>
<li>In the web interface, set the view to two panels using the icon on the top right.</li>
<li>On one side of the interface, click Collection and choose the desired endpoint (<code>chtc#staging</code>, <code>wisc-drive-mzamanian</code>, or your personal computer).</li>
<li>Choose the other endpoint for the other side of the interface.</li>
<li>Type <code>/staging/groups/zamanian_group/</code> into the Path box of the <code>chtc#staging</code> collection and press Enter (you may be required to login with your NetID again).</li>
<li>Navigate to the desired directories.</li>
<li>Drag and drop files to transfer them; you will receive an email upon transfer completion.</li>
<li>Login to the transfer server and archive the directories in staging <code>input/</code> and <code>metadata/</code> with the command <code>tar -cvf {plate}.tar {plate}</code>.
    To loop this command for several files, use: <code>for f in *NJW*; do tar -cvf $f.tar $f; done</code> (use a regular expression that will work for your folders).</li>
<li>Delete the original, unarchived directories.</li>
</ol>
</details>
</li>
<li>
<p><strong>Command line transfer with smbclient</strong>: It is also possible to transfer via smbclient using the terminal. The following code will also archive the data upon arrival.</p>
<details>
<summary>ResearchDrive -&gt; CHTC transfer of unarchived raw data folder (archived on arrival)</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into transfer server and navigate to staging input dir</span>
ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span><span class="w"> </span>/staging/groups/zamanian_group/input/

<span class="c1"># Example of transferring sequencing data</span>
smbclient<span class="w"> </span>-k<span class="w"> </span>//research.drive.wisc.edu/mzamanian/<span class="w"> </span>-D<span class="w"> </span><span class="s2">&quot;UWBC-Dropbox/Bioinformatics Resource Center&quot;</span><span class="w"> </span>-Tc<span class="w"> </span>201105_AHLVWJDSXY.tar<span class="w"> </span><span class="s2">&quot;201105_AHLVWJDSXY&quot;</span>

<span class="c1"># Example of transferring ImageXpress data</span>
smbclient<span class="w"> </span>-k<span class="w"> </span>//research.drive.wisc.edu/mzamanian/<span class="w"> </span>-D<span class="w"> </span><span class="s2">&quot;ImageXpress/raw&quot;</span><span class="w"> </span>-Tc<span class="w"> </span><span class="m">20201118</span>-p01-MZ_172.tar<span class="w"> </span><span class="s2">&quot;20201118-p01-MZ_172.tar&quot;</span>
</code></pre></div>
</details>
<details>
<summary>ResearchDrive -&gt; CHTC transfer of unarchived metadata folder (archived on arrival)</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Log into transfer server and navigate to staging metadata dir</span>
ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span><span class="w"> </span>/staging/groups/zamanian_group/metadata/

<span class="c1"># Example of transferring ImageXpress metadata</span>
smbclient<span class="w"> </span>-k<span class="w"> </span>//research.drive.wisc.edu/mzamanian/<span class="w"> </span>-D<span class="w"> </span><span class="s2">&quot;ImageXpress/metadata&quot;</span><span class="w"> </span>-Tc<span class="w"> </span><span class="m">20201118</span>-p01-MZ_172.tar<span class="w"> </span><span class="s2">&quot;20201118-p01-MZ_172&quot;</span>
</code></pre></div>
</details>
<p><strong>Note:</strong> If you correctly updated your <code>~/.bash_profile</code> by following the <a href="../comp_local/">macOS environment setup instructions</a>, then you can use the simple <code>transfer</code> to <code>ssh</code> into the node.</p>
<p>For ImageXpress data, an entire experiment may include &gt;10 plates that will take hours to days to transfer. To facilitate batch transfers, we have included two scripts in the <code>input/</code> and <code>metadata/</code> directories of <code>/staging/groups/zamanian_group/</code> to help with batch transfer.</p>
<details>
<summary>Batch transfer via custom scripts and <code>screen</code></summary>
<div class="tabbed-set" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><label for="__tabbed_1_1">Instructions</label><div class="tabbed-content">
<ol>
<li>Use a terminal text editor (e.g., <code>vim</code> or <code>nano</code>) to edit the plate list in <code>/staging/groups/zamanian_group/input/plates.txt</code> such that there is a    single plate name per line. Also include a blank line at the end of the file.</li>
<li>Use the commands <code>sh transfer_images.sh</code> (from the <code>input/</code> directory) and <code>sh transfer_metadata.sh</code> (from the <code>metadata/</code> directory) to transfer the images and metadata, respectively.</li>
<li>Use the <code>screen</code> tool to maintain a continuous process in the background, allowing you to close your SSH session.</li>
</ol>
</div>
<input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><label for="__tabbed_1_2">Sample screen commands</label><div class="tabbed-content">
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span>
<span class="normal">11</span>
<span class="normal">12</span>
<span class="normal">13</span>
<span class="normal">14</span>
<span class="normal">15</span>
<span class="normal">16</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="c1"># Log into transfer server and navigate to staging input dir</span>
ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span><span class="w"> </span>/staging/groups/zamanian_group/input/

<span class="c1"># Start a screen named &#39;transfer&#39;</span>
screen<span class="w"> </span>-S<span class="w"> </span>transfer

<span class="c1"># Initiate the transfer</span>
sh<span class="w"> </span>transfer_images.sh

<span class="c1"># Detach from the screen by pressing Ctrl+a and then d</span>
<span class="c1"># Reattach to the screen</span>
screen<span class="w"> </span>-r<span class="w"> </span>transfer

<span class="c1"># Close the screen</span>
<span class="nb">exit</span>
</code></pre></div></td></tr></table></div>
</div>
</div>
</details>
</li>
<li>
<p><strong>Drag-and-drop transfer</strong></p>
<p>Transferring input data should be performed with Globus or background smblient processes. However, smaller files such as metadata or auxiliary parameter files can be moved to the staging directory or one's home directory on the submit server with drag-and-drop SFTP clients such as Transmit, CyberDuck, or FileZilla.</p>
</li>
<li>
<p><strong>Command line transfer with <code>scp</code></strong></p>
<p>A final option for transfer of small files is via the <code>scp</code> command:</p>
<details>
<summary>ResearchDrive -&gt; CHTC transfer of unarchived metadata folder (archived on arrival)</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># transfer to staging</span>
scp<span class="w"> </span><span class="o">[</span><span class="nb">local</span><span class="w"> </span>file<span class="o">]</span><span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu:/staging/groups/zamanian_group/input/

<span class="c1"># transfer to home</span>
scp<span class="w"> </span><span class="o">[</span><span class="nb">local</span><span class="w"> </span>file<span class="o">]</span><span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@ap2002.chtc.wisc.edu:/home/<span class="o">{</span>net-id<span class="o">}</span>
</code></pre></div>
</details>
</li>
</ul>
<details class="danger" open="open">
<summary>Warning!</summary>
<p>Do not use the last two options (drag-and-drop or scp) for the transfer of many files or large files &gt;100 MB, as these methods are unstable and can be unpredictable. For batch transfer or transfer of large files, the recommended method is Globus.</p>
</details>
<h3 id="2-submit-manage-chtc-jobs">2. Submit (manage CHTC jobs)</h3>
<p>CHTC uses HTCondor for job scheduling. Submission files should follow lab conventions and be consistent with the CHTC documentation. Two example submit scripts with annotations are shown below. This submit scripts (Core_RNAseq-nf.sub/wrmXpress.sub) load a pre-defined Docker environment and run a bash executable script (Core_RNAseq-nf.sh/wrmXpress.sh) with defined arguments on the execute node. Other options define log files, resource requirements, and transfer of files in/out of <code>home</code>. Large files should not be transferred in/out of <code>home</code>, but you may need to transfer auxiliary files (for example, the parameters YAML file for wrmXpress jobs). We transfer in our large data through <code>/staging/groups/zamanian_group/input/</code> and we move job output files to <code>/staging/groups/zamanian_group/output/</code> within the job executable script to avoid their transfer to <code>home</code> upon job completion. The only files that should be transferred back to <code>home</code> are small log files.</p>
<details>
<summary>Example CHTC job submission scripts (.sub / .sh)</summary>
<div class="tabbed-set" data-tabs="2:4"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><label for="__tabbed_2_1">Core_RNAseq-nf.sub</label><div class="tabbed-content">
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span>
<span class="normal">11</span>
<span class="normal">12</span>
<span class="normal">13</span>
<span class="normal">14</span>
<span class="normal">15</span>
<span class="normal">16</span>
<span class="normal">17</span>
<span class="normal">18</span>
<span class="normal">19</span>
<span class="normal">20</span>
<span class="normal">21</span>
<span class="normal">22</span>
<span class="normal">23</span>
<span class="normal">24</span>
<span class="normal">25</span>
<span class="normal">26</span>
<span class="normal">27</span>
<span class="normal">28</span>
<span class="normal">29</span>
<span class="normal">30</span>
<span class="normal">31</span>
<span class="normal">32</span>
<span class="normal">33</span>
<span class="normal">34</span></pre></div></td><td class="code"><div><pre><span></span><code># Core_RNAseq-nf.sub
# Input data in /staging/{net-id}/input/$(dir)
# Run: condor_submit Core_RNAseq-nf.sub dir=191211_AHMMC5DMXX script=Core_RNAseq-nf.sh

# request Zamanian Lab server
Accounting_Group = PathobiologicalSciences_Zamanian

# load docker image; request execute server with staging
universe = docker
docker_image = zamanianlab/chtc-rnaseq:v1
Requirements = (Target.HasCHTCStaging == true)

# executable (/home/{net-id}/) and arguments
executable = $(script)
arguments = $(dir)

# log, error, and output files
log = $(dir)_$(Cluster)_$(Process).log
error = $(dir)_$(Cluster)_$(Process).err
output = $(dir)_$(Cluster)_$(Process).out

# transfer files in-out of /home/{net-id}/
transfer_input_files =
should_transfer_files = YES
when_to_transfer_output = ON_EXIT

# memory, disk and CPU requests
request_cpus = 80
request_memory = 500GB
request_disk = 1500GB

# submit 1 job
queue 1
### END
</code></pre></div></td></tr></table></div>
</div>
<input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><label for="__tabbed_2_2">Core_RNAseq-nf.sh</label><div class="tabbed-content">
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span>
<span class="normal">11</span>
<span class="normal">12</span>
<span class="normal">13</span>
<span class="normal">14</span>
<span class="normal">15</span>
<span class="normal">16</span>
<span class="normal">17</span>
<span class="normal">18</span>
<span class="normal">19</span>
<span class="normal">20</span>
<span class="normal">21</span>
<span class="normal">22</span>
<span class="normal">23</span>
<span class="normal">24</span>
<span class="normal">25</span>
<span class="normal">26</span>
<span class="normal">27</span>
<span class="normal">28</span>
<span class="normal">29</span>
<span class="normal">30</span>
<span class="normal">31</span>
<span class="normal">32</span>
<span class="normal">33</span>
<span class="normal">34</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="ch">#!/bin/bash</span>

<span class="c1"># set home () and mk dirs</span>
<span class="nb">export</span><span class="w"> </span><span class="nv">HOME</span><span class="o">=</span><span class="nv">$PWD</span>
mkdir<span class="w"> </span>input<span class="w"> </span>work<span class="w"> </span>output

<span class="c1"># echo core, thread, and memory</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;CPU threads: </span><span class="k">$(</span>grep<span class="w"> </span>-c<span class="w"> </span>processor<span class="w"> </span>/proc/cpuinfo<span class="k">)</span><span class="s2">&quot;</span>
grep<span class="w"> </span><span class="s1">&#39;cpu cores&#39;</span><span class="w"> </span>/proc/cpuinfo<span class="w"> </span><span class="p">|</span><span class="w"> </span>uniq
<span class="nb">echo</span><span class="w"> </span><span class="k">$(</span>free<span class="w"> </span>-g<span class="k">)</span>

<span class="c1"># transfer input data from staging ($1 is ${dir} from args)</span>
cp<span class="w"> </span>-r<span class="w"> </span>/staging/groups/zamanian_group/input/<span class="nv">$1</span>.tar<span class="w"> </span>input
<span class="nb">cd</span><span class="w"> </span>input<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>tar<span class="w"> </span>-xvf<span class="w"> </span><span class="nv">$1</span>.tar<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>rm<span class="w"> </span><span class="nv">$1</span>.tar<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>mv<span class="w"> </span>*/*/*<span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>..

<span class="c1"># clone nextflow git repo</span>
git<span class="w"> </span>clone<span class="w"> </span>https://github.com/zamanianlab/Core_RNAseq-nf.git

<span class="c1"># run nextflow command</span>
<span class="nb">export</span><span class="w"> </span><span class="nv">NXF_OPTS</span><span class="o">=</span><span class="s1">&#39;-Xms1g -Xmx8g&#39;</span>
nextflow<span class="w"> </span>run<span class="w"> </span>Core_RNAseq-nf/WB-pe.nf<span class="w"> </span>-w<span class="w"> </span>work<span class="w"> </span>-c<span class="w"> </span>Core_RNAseq-nf/chtc.config<span class="w"> </span>--dir<span class="w"> </span><span class="nv">$1</span><span class="se">\</span>
<span class="w">    </span>--star<span class="w"> </span>--qc<span class="w"> </span>--release<span class="w"> </span><span class="s2">&quot;WBPS15&quot;</span><span class="w"> </span>--species<span class="w"> </span><span class="s2">&quot;brugia_malayi&quot;</span><span class="w"> </span>--prjn<span class="w"> </span><span class="s2">&quot;PRJNA10729&quot;</span><span class="w"> </span>--rlen<span class="w"> </span><span class="s2">&quot;150&quot;</span>

<span class="c1"># rm files you don&#39;t want transferred back to /home/{net-id}</span>
rm<span class="w"> </span>-r<span class="w"> </span>work<span class="w"> </span>input

<span class="c1"># tar output folder and delete it</span>
<span class="nb">cd</span><span class="w"> </span>output<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>tar<span class="w"> </span>-cvf<span class="w"> </span><span class="nv">$1</span>.tar<span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>rm<span class="w"> </span>-r<span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>..

<span class="c1"># remove staging output tar if there from previous run</span>
rm<span class="w"> </span>-f<span class="w"> </span>/staging/groups/zamanian_group/output/<span class="nv">$1</span>.tar

<span class="c1"># mv large output files to staging output folder; avoid their transfer back to /home/{net-id}</span>
mv<span class="w"> </span>output/<span class="nv">$1</span>.tar<span class="w"> </span>/staging/groups/zamanian_group/output/
</code></pre></div></td></tr></table></div>
</div>
<input id="__tabbed_2_3" name="__tabbed_2" type="radio" /><label for="__tabbed_2_3">wrmXpress.sub</label><div class="tabbed-content">
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span>
<span class="normal">11</span>
<span class="normal">12</span>
<span class="normal">13</span>
<span class="normal">14</span>
<span class="normal">15</span>
<span class="normal">16</span>
<span class="normal">17</span>
<span class="normal">18</span>
<span class="normal">19</span>
<span class="normal">20</span>
<span class="normal">21</span>
<span class="normal">22</span>
<span class="normal">23</span>
<span class="normal">24</span>
<span class="normal">25</span>
<span class="normal">26</span>
<span class="normal">27</span>
<span class="normal">28</span>
<span class="normal">29</span>
<span class="normal">30</span>
<span class="normal">31</span>
<span class="normal">32</span>
<span class="normal">33</span>
<span class="normal">34</span></pre></div></td><td class="code"><div><pre><span></span><code># Input data: /staging/groups/zamanian_group/input/${plate}.tar
# Parameters: $HOME/${plate}.yml
# Run: condor_submit wrmXpress.sub script=wrmXpress.sh

# request Zamanian Lab server
Accounting_Group = PathobiologicalSciences_Zamanian

# load docker image; request execute server with large data staging
universe = docker
docker_image = zamanianlab/chtc-wrmxpress:v1
Requirements = (Target.HasCHTCStaging == true)

# executable (/home/{net-id}/) and arguments
executable = $(script)
arguments = $(plate)

# log, error, and output files
log = $(plate)_$(Cluster)_$(Process).log
error = $(plate)_$(Cluster)_$(Process).err
output = $(plate)_$(Cluster)_$(Process).out

# transfer files in-out of /home/{net-id}/
transfer_input_files = $(plate).yml
should_transfer_files = YES
when_to_transfer_output = ON_EXIT

# memory, disk and CPU requests
request_cpus = 1
request_memory = 8GB
request_disk = 10GB

# submit a job for each directory in plate_list.txt
queue plate from plate_list.txt
### END
</code></pre></div></td></tr></table></div>
</div>
<input id="__tabbed_2_4" name="__tabbed_2" type="radio" /><label for="__tabbed_2_4">wrmXpress.sh</label><div class="tabbed-content">
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span>
<span class="normal">11</span>
<span class="normal">12</span>
<span class="normal">13</span>
<span class="normal">14</span>
<span class="normal">15</span>
<span class="normal">16</span>
<span class="normal">17</span>
<span class="normal">18</span>
<span class="normal">19</span>
<span class="normal">20</span>
<span class="normal">21</span>
<span class="normal">22</span>
<span class="normal">23</span>
<span class="normal">24</span>
<span class="normal">25</span>
<span class="normal">26</span>
<span class="normal">27</span>
<span class="normal">28</span>
<span class="normal">29</span>
<span class="normal">30</span>
<span class="normal">31</span>
<span class="normal">32</span>
<span class="normal">33</span>
<span class="normal">34</span>
<span class="normal">35</span>
<span class="normal">36</span>
<span class="normal">37</span>
<span class="normal">38</span>
<span class="normal">39</span>
<span class="normal">40</span>
<span class="normal">41</span>
<span class="normal">42</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="ch">#!/bin/bash</span>

<span class="c1"># set home () and mk dirs</span>
<span class="nb">export</span><span class="w"> </span><span class="nv">HOME</span><span class="o">=</span><span class="nv">$PWD</span>
mkdir<span class="w"> </span>input
mkdir<span class="w"> </span>metadata
mkdir<span class="w"> </span>output/
mkdir<span class="w"> </span>work/

git<span class="w"> </span>clone<span class="w"> </span>https://github.com/zamanianlab/wrmXpress.git

<span class="c1"># echo core, thread, and memory</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;CPU threads: </span><span class="k">$(</span>grep<span class="w"> </span>-c<span class="w"> </span>processor<span class="w"> </span>/proc/cpuinfo<span class="k">)</span><span class="s2">&quot;</span>
grep<span class="w"> </span><span class="s1">&#39;cpu cores&#39;</span><span class="w"> </span>/proc/cpuinfo<span class="w"> </span><span class="p">|</span><span class="w"> </span>uniq
<span class="nb">echo</span><span class="w"> </span><span class="k">$(</span>free<span class="w"> </span>-g<span class="k">)</span>

<span class="c1"># transfer and decompress input data from staging ($1 is ${dir} from args)</span>
tar<span class="w"> </span>-xf<span class="w"> </span>/staging/groups/zamanian_group/input/<span class="nv">$1</span>.tar<span class="w"> </span>-C<span class="w"> </span>input/

<span class="c1"># deprecated</span>
<span class="c1"># cp -r /staging/groups/zamanian_group/input/$1.tar input</span>
<span class="c1"># cd $input &amp;&amp; tar --strip-components 5 -xvf $1.tar &amp;&amp; cd $HOME</span>

<span class="c1"># transfer and decompress metadata from staging ($1 is ${dir} from args)</span>
tar<span class="w"> </span>-xf<span class="w"> </span>/staging/groups/zamanian_group/metadata/<span class="nv">$1</span>.tar<span class="w"> </span>-C<span class="w"> </span>metadata

<span class="c1"># deprecated</span>
<span class="c1"># cd metadata &amp;&amp; tar -xvf $1.tar &amp;&amp; rm $1.tar &amp;&amp; mv */*/* $1 &amp;&amp; cd $HOME</span>

<span class="c1"># run the wrapper</span>
python<span class="w"> </span>Core_imgproc/wrapper.py<span class="w"> </span><span class="nv">$1</span>.yml<span class="w"> </span><span class="nv">$1</span>

<span class="c1"># tar output folder and delete it</span>
mv<span class="w"> </span>output<span class="w"> </span><span class="nv">$1</span>
mv<span class="w"> </span><span class="nv">$1</span>.yml<span class="w"> </span><span class="nv">$1</span>
tar<span class="w"> </span>-cvf<span class="w"> </span><span class="nv">$1</span>.tar<span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>rm<span class="w"> </span>-r<span class="w"> </span><span class="nv">$1</span>

<span class="c1"># remove staging output tar if there from previous run</span>
rm<span class="w"> </span>-f<span class="w"> </span>/staging/groups/zamanian_group/output/<span class="nv">$1</span>.tar

<span class="c1"># mv large output files to staging output folder; avoid their transfer back to /home/{net-id}</span>
mv<span class="w"> </span><span class="nv">$1</span>.tar<span class="w"> </span>/staging/groups/zamanian_group/output/
</code></pre></div></td></tr></table></div>
</div>
</div>
</details>
<p>Log into submit node to submit a job,</p>
<div class="highlight"><pre><span></span><code>ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@ap2002.chtc.wisc.edu

condor_submit<span class="w"> </span>Core_RNAseq-nf.sub<span class="w"> </span><span class="nv">dir</span><span class="o">=</span>191211_AHMMC5DMXX<span class="w"> </span><span class="nv">script</span><span class="o">=</span>Core_RNAseq-nf.sh
</code></pre></div>
<details>
<summary>Other useful commands for monitoring and managing jobs</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># check on job status</span>
<span class="w">  </span>condor_q

<span class="c1"># remove a specific job</span>
<span class="w">  </span>condor_rm<span class="w"> </span><span class="o">[</span>job<span class="w"> </span>id<span class="o">]</span>

<span class="c1"># remove all jobs for user</span>
<span class="w">  </span>condor_rm<span class="w"> </span><span class="nv">$USER</span>

<span class="c1"># interative shell to running job on remote machine</span>
<span class="w">  </span>condor_ssh_to_job<span class="w"> </span><span class="o">[</span>job<span class="w"> </span>id<span class="o">]</span>
<span class="w">  </span><span class="nb">exit</span>
</code></pre></div>
</details>
<h3 id="3-output-chtc-researchdrive">3. Output (CHTC &rarr; ResearchDrive)</h3>
<p>To transfer your job output folder from the CHTC staging output directory to ResearchDrive or your desktop, it is easiest to use Globus or an SFTP client (e.g., CyberDuck). It also possible to use the command line to transfer outputs to ResearchDrive or your desktop.</p>
<details>
<summary>CHTC -&gt; ResearchDrive transfer with smbclient</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># log into CHTC staging server and navigate to output folder</span>
ssh<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu
<span class="nb">cd</span><span class="w"> </span>/staging/groups/zamanian_group/output/

<span class="c1"># connect to lab ResearchDrive</span>
smbclient<span class="w"> </span>-k<span class="w"> </span>//research.drive.wisc.edu/mzamanian

<span class="c1"># turn off prompting and turn on recursive</span>
smb:<span class="w"> </span><span class="se">\&gt;</span><span class="w"> </span>prompt
smb:<span class="w"> </span><span class="se">\&gt;</span><span class="w"> </span>recurse

<span class="c1"># navigate to ResearchDrive dir for processed data (example)</span>
smb:<span class="w"> </span><span class="se">\&gt;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>/ImageXpress/proc/

<span class="c1"># transfer output data folder (example)</span>
smb:<span class="w"> </span><span class="se">\&gt;</span><span class="w"> </span>mput<span class="w"> </span><span class="m">20201119</span>-p01-MZ_200.tar
</code></pre></div>
</details>
<details>
<summary>CHTC -&gt; Local Computer</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># transfer from CHTC staging to current directory</span>
scp<span class="w"> </span>-r<span class="w"> </span><span class="o">{</span>net-id<span class="o">}</span>@transfer.chtc.wisc.edu:/staging/groups/zamanian_group/output/<span class="o">[</span>dir<span class="o">]</span><span class="w"> </span>.
</code></pre></div>
</details>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["instant"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.fe8b6f2b.min.js"></script>
      
        <script src="../javascripts/extra.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      
    
  </body>
</html>